src/rustllvm/PassWrapper.cpp

```diff
@@ -556,7 +556,8 @@ LLVMRustWriteOutputFile(LLVMTargetMachineRef Target, LLVMPassManagerRef PMR,
   }
 
 #if LLVM_VERSION_GE(7, 0)
-  unwrap(Target)->addPassesToEmitFile(*PM, OS, nullptr, FileType, false);
+  buffer_ostream BOS(OS);
```

Comment:
Is there a reason this is only done on 7.0+? Seems like we could do it for other LLVMs as well.

Reply:
I didn't want to muck around with older versions without testing, so I opted to leave it here (it doesn't really matter for older llvm, a slightly slower compiler is the least of those worries)